### ISSUE TITLE
We need a sitemap so that Google will index our site more quickly

### ISSUE THREAD

**Issue by salimfadhley** [pythonjobs/jobs#35]
https://support.google.com/webmasters/answer/183668?hl=en&authuser=0&ref_topic=6080646&rd=1

**Comment by salimfadhley** [comment]
I think we can use the same code that makes the ATOM feed to do this, right? It's virtually the same thing.

**Comment by salimfadhley** [comment]
https://github.com/lakshmivyas/hyde/wiki/Site-Post-Processors

This recipe does not seem to correspond to anything we actually have in the site.

**Comment by stestagg** [comment]
Yeah, I've just read the spec again for sitemaps, to check (
https://support.google.com/webmasters/answer/183668?hl=en&ref_topic=6080646&rd=1)
I see absolutely no reason for a custom sitemap plugin for this, we can
just generate it like the atom file.

How I would do this:

in content/jobs/meta.yaml, add an item:  sitemap-priority:0.5
in index.html, add sitemap-priority: 1.0
(any other files (excluding the jobs) we want in the sitemap should also
have a priority.

Then add a sorter to the site.yaml file, called priority

Then, in a new sitemap.xml file in content/, do something like (this is off
my head, so might be wrong):

<xml...
{% for resource in site.walk_resources_sorted_by_priority %}
{% if resource.meta.sitemap_priority %}
<url>...
{% endif %}
{% endfor %}
.../urlset>

We can then submit this to google manually whenever we want, or add
something to the deploy script that posts it automatically

**Comment by salimfadhley** [comment]
I did this without the priority business - I just hard-coded the index as the first URL, and then had all the jobs followed by all the tags. Do we really need the prority stuff?

**Comment by stestagg** [comment]
The priority thing was about using the priority to indicate which pages should be promoted more, but that's hardly important, the other thing was it allows us to flag which 'nodes' should be included magically in the sitemap, without putting knowledge of jobs and tags and indexes etc.. in it by hand. (because _any_ node with a priority>0 is included, and the meta cascading handles everything for us)

So basically, it's a nice-to-have, but certainly not needed

**Comment by salimfadhley** [comment]
I like that idea - can we get the basic version live (subject to review) and then I will add it once we have enough content. Apparantly google's sitemaps can support 50k rows so we have plenty of time.
